Tests/_CollectionsTestSupport/AssertionContexts/CollectionTestCase.swift

```diff
@@ -19,7 +19,7 @@ open class CollectionTestCase: XCTestCase {
 
   open var isAvailable: Bool { true }
 
-  #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+  #if canImport(Darwin)
```

Comment:
Ooh, this one is different. 

My worry is that `canImport(Darwin)` might return true if the project being built includes an unrelated module that coincidentally happens to be named `Darwin`. Targets not being cleanly isolated from each other has been a constant source of low-level issues.

I'd prefer if we did this:

```suggestion
  #if _runtime(_ObjC)
```

(Well, what I'd actually prefer is a Swift native test environment that understands availability, but this is a good stand-in. 😛)

Reply:
Awesome! I will close this PR, since this topic needs more clarification. The points you bring here is all valid to me. Now, I'm actually lost 🙈.

Thanks for sharing this!